src/tests/examples/example_test.py

```diff
@@ -36,102 +42,92 @@ def launch_xpansion(install_dir, study_path, method):
 
-    command = [sys.executable, "../../src_python/launch.py", "--installDir", install_dir_full, "--dataDir", str(study_path),"--method", method, "--step","full","-n","2"]
+    command = [sys.executable, "../../src_python/launch.py", "--installDir", install_dir_full, "--dataDir",
+               str(study_path), "--method", method, "--step", "full", "-n", "2"]
     process = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=None)
     output = process.communicate()
 
-    if process.returncode !=0 :
+    if process.returncode != 0:
         log_path = find_log_path(study_path / 'output')
-        #print log
+        # print log
         print(open(str(log_path), 'r').read())
 
     # Check return value
     assert process.returncode == 0
 
 
-def check_investment_solution(study_path, expected_last_it,expected_investment_solution):
+def check_solution(study_path, expected_values, expected_investment_solution):
     output_path = study_path / 'output'
-    log_path = find_log_path(output_path)
-
-    log_file = open(str(log_path), 'r')
-    log = log_file.readline()
-
-    last_it = []
-    investment_solution = {}
-
-    # search in log content for iteration result and investment solution
-    # TODO : it could be better to check the .json result file instead of the log file
-    while log:
-        if "ITE" in log and "LB" in log and "UB" in log:
-            it = log_file.readline().strip()
-            while len(it) != 0:
-                #check if its a iteration log
-                splitStr = it.split()
-                if len(splitStr)==10:
-                    last_it = [float(i) for i in splitStr]
-                it = log_file.readline().strip()
-        if "Investment solution" in log:
-            res = log_file.readline().strip()
-            while len(res) != 0:
-                if ("=" in res):
-                    investment_solution_list = res.replace("|", "").strip().split("=")
-                    assert (len(investment_solution_list) == 2)
-                    investment_solution[investment_solution_list[0].strip()] = float(investment_solution_list[1])
-                res = log_file.readline().strip()
-        log = log_file.readline()
-
-    log_file.close()
-
-    np.testing.assert_allclose(last_it[1:3], expected_last_it[0:2], rtol=1e-4, atol=0)
+    json_path = find_json_output(output_path)
+
+    json_file = open(str(json_path), 'r')
+    json_data = json.load(json_file)
+
+    solution = json_data["solution"]
+    investment_solution = solution["values"]
+
+    np.testing.assert_allclose(solution["gap"], expected_values["gap"], rtol=1e-4, atol=0)
+    np.testing.assert_allclose(solution["investment_cost"], expected_values["investment_cost"], rtol=1e-4, atol=0)
+    np.testing.assert_allclose(solution["operational_cost"], expected_values["operational_cost"], rtol=1e-4, atol=0)
+    np.testing.assert_allclose(solution["overall_cost"], expected_values["overall_cost"], rtol=1e-4, atol=0)
```

Comment:
introduce constant
`RELATIVE_TOLERANCE=1e-4`

`atol=0` by default

```suggestion
    RELATIVE_TOLERANCE=1e-4
    np.testing.assert_allclose(solution["gap"], expected_values["gap"], rtol=RELATIVE_TOLERANCE)
    np.testing.assert_allclose(solution["investment_cost"], expected_values["investment_cost"], rtol=RELATIVE_TOLERANCE)
    np.testing.assert_allclose(solution["operational_cost"], expected_values["operational_cost"], rtol=RELATIVE_TOLERANCE)
    np.testing.assert_allclose(solution["overall_cost"], expected_values["overall_cost"], rtol=RELATIVE_TOLERANCE)
```